src/WebJobs.Extensions.DurableTask/DurableHttpRequest.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Runtime.Serialization;
+using Microsoft.Extensions.Primitives;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Request used to make an HTTP call through Durable Functions.
+    /// </summary>
+    public class DurableHttpRequest
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DurableHttpRequest"/> class.
+        /// </summary>
+        /// <param name="method">Method used for HTTP request.</param>
+        /// <param name="uri">Uri used to make the HTTP request.</param>
+        public DurableHttpRequest(HttpMethod method, Uri uri)
+        {
+            this.Method = method;
+            this.Uri = uri;
+            this.Headers = new Dictionary<string, StringValues>(StringComparer.OrdinalIgnoreCase);
+            this.TokenSource = null;
+        }
+
+        /// <summary>
+        /// HttpMethod used in the HTTP request made by the Durable Function.
+        /// </summary>
+        public HttpMethod Method { get; set; }
+
+        /// <summary>
+        /// Uri used in the HTTP request made by the Durable Function.
+        /// </summary>
+        public Uri Uri { get; set; }
+
+        /// <summary>
+        /// Headers passed with the HTTP request made by the Durable Function.
+        /// </summary>
+        // TODO: Need to change this to IDictionary<string, StringValues>
+        public IDictionary<string, StringValues> Headers { get; set; }
```

Comment:
Let's remove the `set` here. I don't think we want other classes to be able to set this. You can still set it from the constructor even without the `set`. Same with `Uri` and `Method` properties. The less mutable you can make a class, the better (I'm assuming you don't have any cases where you create a `DurableHttpRequest` object and then need to keep manipulating it).

Reply:
I can take the `set` out for `Method`, but `Uri` and `Headers` are set in DurableOrchestrationContext. `Uri` is changed to set the location header of an asynchronous response (line 244 in DurableOrchestrationContext.cs). `Headers` are set when a copy of the DurableHttpRequest is made (line 250 in DurableOrchestrationContext.cs). Also, when users create DurableHttpRequest messages they set all attributes other than `Method `and `Uri` outside of the constructor.

@cgillum Would it be better to make all of the attributes other then `Method `and `Uri `to be optional arguments in the constructor? This would change how the user creates a DurableHttpRequest.

Reply:
I actually ended up removing the set from all of the attributes so the class is immutable and changes can only be made by creating a new DurableHttpRequest.